core/lib/presentation/views/text/rich_text_builder.dart

```diff
@@ -56,12 +120,79 @@ class RichTextBuilder extends StatelessWidget {
       // style the matched text
       final endIndex = startIndex + word.length;
       final spanText = text.substring(startIndex, endIndex);
-      spans.add(TextSpan(text: spanText, style: styleWord));
+
+      if (!firstHighlightKeyed) {
+        spans.add(WidgetSpan(child: Text(
+          spanText,
+          key: firstHighlightKey,
+          style: styleWord,
+        )));
+        firstHighlightKeyed = true;
+      } else {
+        spans.add(TextSpan(text: spanText, style: styleWord));
+      }
       // mark the boundary to start the next search from
       spanBoundary = endIndex;
       // continue until there are no more matches
     } while (spanBoundary < text.length);
 
     return spans;
   }
+
+  List<InlineSpan> _getSpansFromPreMarkedText() {
+    List<InlineSpan> spans = [];
+    String? processingText = widget.preMarkedText;
+    bool firstHighlightKeyed = false;
+    
+    while (processingText != null && processingText.isNotEmpty) {
+      final startIndex = processingText.indexOf(_startMark);
+
+      // There is no more <mark> in the text
+      if (startIndex == -1) {
+        spans.add(TextSpan(text: processingText, style: widget.styleOrigin));
+        return spans;
+      }
+
+      final endIndex = processingText.indexOf(
+        _endMark,
+        startIndex + _startMark.length);
+      // There is start <mark> but no end </mark>
+      if (endIndex == -1) {
```

Comment:
should add widget test for this case too

Reply:
Done